internal/common/platforms.go

```diff
@@ -0,0 +1,37 @@
+package common
+
+import "github.com/openshift/assisted-service/models"
+
+const (
+	VmwareManufacturer string = "VMware, Inc."
+)
+
+var supportedManufacturers = map[string]models.PlatformType{
+	VmwareManufacturer: models.PlatformTypeVsphere,
+}
+
+func AppendPlatformFromVendor(vendor models.SystemVendor, hostsPlatform *[]models.PlatformType) {
+	platform := supportedManufacturers[vendor.Manufacturer]
+	if platform != "" {
+		*hostsPlatform = append(*hostsPlatform, platform)
+	}
+}
+
+func DecideSupportedPlatforms(hostsPlatforms []models.PlatformType, hostsCount int) *[]models.PlatformType {
+	supportedPlatforms := []models.PlatformType{models.PlatformTypeBaremetal}
+
+	// SNO or no hosts
+	if hostsCount == 0 || hostsCount == 1 || len(hostsPlatforms) < hostsCount {
```

Comment:
SNO should be determined by cluster.high_availability_mode == "None" (i.e., the user chose a non-highly-available control plane).  One host could just mean that not enough hosts have registered so far.